aviary/subsystems/propulsion/engine_deck.py

```diff
@@ -1010,24 +1008,30 @@ def build_mission(self, num_nodes, aviary_inputs) -> om.Group:
 
         return engine_group
 
+    def get_parameters(self):
+        params = {}
+        params[Aircraft.Engine.SCALE_FACTOR] = {'static_target': True}
+        return params
+
     def report(self, problem, reports_file, **kwargs):
         meta_data = kwargs['meta_data']
+        engine_idx = kwargs['engine_idx']
 
         outputs = [Aircraft.Engine.NUM_ENGINES,
                    Aircraft.Engine.SCALED_SLS_THRUST,
                    Aircraft.Engine.SCALE_FACTOR]
 
         # determine which index in problem-level aviary values corresponds to this engine
-        engine_idx = None
-        for idx, engine in enumerate(problem.aviary_inputs.get_val('engine_models')):
-            if engine.name == self.name:
-                engine_idx = idx
+        # engine_idx = None
```

Comment:
You an remove these commented out lines now that the engine number is passed into this report.